drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/util/DMNSimulationSettingsCreationStrategy.java

```diff
@@ -71,15 +71,46 @@ public Simulation createSimulation(Path context, String dmnFilePath) throws Exce
             return aType.equals(bType) ? 0 : (Type.INPUT.equals(aType) ? -1 : 1);
         }).forEach(factModelTree -> {
             FactIdentifier factIdentifier = new FactIdentifier(factModelTree.getFactName(), factModelTree.getFactName());
-            FactMappingExtractor factMappingExtractor = new FactMappingExtractor(factIdentifier, scenarioWithIndex.getIndex(), id, convert(factModelTree.getType()), simulationDescriptor, scenarioWithIndex.getScenario());
-            addToScenario(factMappingExtractor, factModelTree, new ArrayList<>(), hiddenValues);
+            FactMappingExtractor factMappingExtractor = new FactMappingExtractor(factIdentifier, scenarioWithIndex.getIndex(), id, convert(factModelTree.getType()), simulationDescriptor, scenarioWithIndex.getScesimData());
+            addFactMapping(factMappingExtractor, factModelTree, new ArrayList<>(), hiddenValues);
```

Comment:
@jomarko 
You are probably right, but I did not change that part - I only renamed the called method. So (being already in master) I would leave it as it is for the moment being (I would prefer to not add other "variables" to this already longish PR) - it is another think to look again ASAP

Reply:
@gitgabrio @jomarko 
If I remember correctly I have recently introduced the nested `new ArrayList<>(...)` and I forgot to reconsider the external call so I guess it should be possible to change to `Collections.emptyList()`. I would like not to change it now because I don't like to change caller code based on an implementation detail (that may change). If we really want to do it I agree to do it in another PR and also add javadoc + test + variable rename to _ensure_ the immutability of that parameter